cmd/checkpoint/main.go

```diff
@@ -0,0 +1,268 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/v1"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/release_1_2"
+	"k8s.io/kubernetes/pkg/client/restclient"
+	"k8s.io/kubernetes/pkg/client/unversioned/clientcmd"
+)
+
+var manifestBytes = []byte(`
+{
+"apiVersion": "v1",
+"kind": "Pod",
+"metadata": {
+  "name": "temp-apiserver",
+  "namespace": "kube-system"
+},
+"spec": {}
+}
+`)
+
+const (
+	kubeletAPIPodsURL = "http://localhost:10255/pods"
+	ignorePath        = "/srv/kubernetes/manifests"
+	activePath        = "/etc/kubernetes/manifests"
+	manifestFilename  = "apiserver.json"
+	kubeconfigPath    = "/etc/kubernetes/kubeconfig"
+	tlsPath           = "/etc/kubernetes/tls"
+)
+
+var (
+	tempAPIServer = []byte("temp-apiserver")
+	kubeAPIServer = []byte("kube-apiserver")
+	activeFile    = filepath.Join(activePath, manifestFilename)
+	ignoreFile    = filepath.Join(ignorePath, manifestFilename)
+	secureAPIAddr = fmt.Sprintf("https://%s:%s", os.Getenv("KUBERNETES_SERVICE_HOST"), os.Getenv("KUBERNETES_SERVICE_PORT_HTTPS"))
+)
+
+func main() {
+	log.Println("begin apiserver checkpointing...")
+	run()
+}
+
+func run() {
+	var tempAPIServerManifest v1.Pod
+	if err := json.Unmarshal(manifestBytes, &tempAPIServerManifest); err != nil {
```

Comment:
Initially I had that because I wasn't using any upstream objects (as data structures) but now that those are more prevalent here I think it makes sense to use a v1.Pod object, for sure.